projects/power_distribution/src/publish_data.c

```diff
@@ -19,17 +30,42 @@ StatusCode power_distribution_publish_data_init(PowerDistributionPublishConfig c
   return STATUS_CODE_OK;
 }
 
+static void prv_partially_publish(SoftTimerId timer_id, void *context) {
+  if (s_current_measurements == NULL) {
+    // guard against possible segfaults
+    return;
+  }
+
+  // We use a uintptr_t for the index so we can pass it through the context
+  uintptr_t index = (uintptr_t)context;
+
+  uint16_t transmit_up_to = MIN(index + NUM_MSG_PER_ITERATION, s_config.num_currents_to_publish);
+  while (index < transmit_up_to) {
+    PowerDistributionCurrent current_id = s_config.currents_to_publish[index];
+    StatusCode tx_code = s_config.transmitter(current_id, s_current_measurements[current_id]);
+    if (!status_ok(tx_code)) {
```

Comment:
Is there any other error handling needed here? The previous method called status_ok_or_return on each tx. There isn't any handling in main.c set up right now i guess so this shouldn't be an issue, but would it be a good idea to maybe re-transmit?

Reply:
The messages aren't critical and it'll be retransmitted in the next cycle in 500ms anyways, so I figure it's not that big a deal. I prefer this version to the old one anyways - the previous version silently ignored errors and just didn't tx the rest of the currents, whereas this one at least tries and alerts us.